cfgov/unprocessed/js/organisms/Carousel.js

```diff
@@ -80,17 +89,61 @@ function Carousel( element ) {
         ++i;
       }
     }
-    if ( _currItemIndex !== i ) {
-      _updateDisplay( _currItemIndex, i );
+
+    if ( _updateDisplay( _currItemIndex, i ) ) {
+      _sendAnalytics( i, 'Tab' );
+    }
+  }
+
+  /**
+   * Sends events to our analytics provider.
+   *
+   * @param {number} newCurrItemIndex - The new index value of the selected item.
+   * @param {string} interactionLabel - A label for what called this method.
+   */
+  function _sendAnalytics( newCurrItemIndex, interactionLabel ) {
+    // Send individual event
+    Analytics.sendEvent( {
+      event: 'Page Interaction',
+      action: 'Carousel',
+      label: `${ interactionLabel } Clicked: Page ${ _currItemIndex + 1 }`
+    } );
+
+    /**
+     * Send All Items Seen event if all slides have been visited.
+     *
+     * We use a bitmask to see if all slides have been visited:
+     * 0001 = item one (1 decimal).
+     * 0010 = item two (2 decimal).
+     * 0100 = item three (4 decimal).
+     * 1000 = item four (8 decimal).
+     * 1111 = all visited (15 decimal).
+     *
+     * We need to map 0, 1, 2, 3 item indices to bitwise digits to 1, 2, 4, 8.
+     */
+    _slideVisitCount |= [ 1, 2, 4, 8 ][newCurrItemIndex];
```

Comment:
Would this be more future-proof to a variable number of slides?

```suggestion
    _slideVisitCount |= Math.pow( 2, newCurrItemIndex );
```

Then you'd also want to remove the hardcoded `ALL_SLIDES_VISITED` and instead compare against `Math.pow( 2, _items.length ) - 1` (or something similar).

Reply:
That makes sense. I'll try it after we sort out this other stuff.

Reply:
Updated! Great suggestion; thanks.